### PR TITLE
Support certificate content in Opensearch Source configuration to sup…

### PR DESCRIPTION
**Pull request by dinujoh**
Support certificate content in Opensearch Source configuration to support TLS and Truststore for Opensearch client.
 
### Check List
- [x] New functionality includes testing.
- [ ] New functionality has a documentation issue. Please link to it in this PR.
  - [ ] New functionality has javadoc added
- [x] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/configuration/ConnectionConfiguration.java [hunk 1]
```diff
@@ -14,6 +14,9 @@ public class ConnectionConfiguration {
   @JsonProperty("cert")
   private Path certPath;
 
+  @JsonProperty("certificate_content")
```
Comment (dlvenable) [inline data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/configuration/ConnectionConfiguration.java:17]: Should we have a validation on mutual exclusivity?

```
@AssertTrue
public boolean certificateFileAndContentAreMutuallyExclusive() {
  if(certPath == null && certificateContent == null)
    return true;
  return certPath != null ^ certificateContent != null;
}
```

**Review by dlvenable — DISMISSED** [review]

File: data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/configuration/ConnectionConfiguration.java (hunk 1 above)
Comment (dinujoh) [reply data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/configuration/ConnectionConfiguration.java:18]: Done

File: data-prepper-plugins/opensearch/src/test/java/org/opensearch/dataprepper/plugins/source/opensearch/worker/client/OpenSearchClientFactoryTest.java [hunk 2]
```diff
@@ -209,4 +217,59 @@ void provideOpenSearchClient_with_aws_auth_and_serverless_flag_true() {
...
+            assertThat(elasticsearchClient, notNullValue());
+            trustStoreProviderMockedStatic.verify(() -> TrustStoreProvider.createSSLContext(path));
+        }
+    }
+
+
+    @Test
+    void createSdkAsyncHttpClient_with_self_signed_certificate() {
+        try (MockedStatic<TrustStoreProvider> trustStoreProviderMockedStatic = mockStatic(TrustStoreProvider.class)) {
+            final Path path = mock(Path.class);
```
[the remark this answers is not in the document]
Comment (dinujoh) [reply data-prepper-plugins/opensearch/src/test/java/org/opensearch/dataprepper/plugins/source/opensearch/worker/client/OpenSearchClientFactoryTest.java]: Done

File: data-prepper-plugins/opensearch/src/test/java/org/opensearch/dataprepper/plugins/source/opensearch/worker/client/OpenSearchClientFactoryTest.java [hunk 3]
```diff
@@ -209,4 +217,59 @@ void provideOpenSearchClient_with_aws_auth_and_serverless_flag_true() {
...
+            final OpenSearchClient openSearchClient = createObjectUnderTest().provideOpenSearchClient(openSearchSourceConfiguration);
+            assertThat(openSearchClient, notNullValue());
+            trustStoreProviderMockedStatic.verify(() -> TrustStoreProvider.createSSLContext(path));
+        }
+    }
+
+    @Test
+    void provideElasticSearchClient_with_self_signed_certificate() {
+        try (MockedStatic<TrustStoreProvider> trustStoreProviderMockedStatic = mockStatic(TrustStoreProvider.class)) {
+            final Path path = mock(Path.class);
```
[the remark this answers is not in the document]
Comment (dinujoh) [reply data-prepper-plugins/opensearch/src/test/java/org/opensearch/dataprepper/plugins/source/opensearch/worker/client/OpenSearchClientFactoryTest.java]: Done

File: data-prepper-plugins/opensearch/src/test/java/org/opensearch/dataprepper/plugins/source/opensearch/worker/client/OpenSearchClientFactoryTest.java [hunk 4]
```diff
@@ -209,4 +217,59 @@ void provideOpenSearchClient_with_aws_auth_and_serverless_flag_true() {
         assertThat(awsCredentialsOptions.getStsHeaderOverrides(), equalTo(Collections.emptyMap()));
         assertThat(awsCredentialsOptions.getStsRoleArn(), equalTo(stsRoleArn));
     }
+
+    @Test
+    void provideOpenSearchClient_with_self_signed_certificate() {
+        try (MockedStatic<TrustStoreProvider> trustStoreProviderMockedStatic = mockStatic(TrustStoreProvider.class)) {
+            final Path path = mock(Path.class);
```
[the remark this answers is not in the document]
Comment (dinujoh) [reply data-prepper-plugins/opensearch/src/test/java/org/opensearch/dataprepper/plugins/source/opensearch/worker/client/OpenSearchClientFactoryTest.java]: Done

**Review by dlvenable — APPROVED** [review]

**Review by kkondaka — APPROVED** [review]

**Pull request merged**